cmd/publicize/server.go

```diff
@@ -0,0 +1,237 @@
+package main
+
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"net/url"
+	"os"
+	"os/exec"
+	"regexp"
+	"strings"
+
+	"github.com/sirupsen/logrus"
+
+	prowconfig "k8s.io/test-infra/prow/config"
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/pluginhelp"
+	"k8s.io/test-infra/prow/plugins"
+)
+
+var publicizeRe = regexp.MustCompile(`(?mi)^/publicize\s*$`)
+
+func helpProvider(_ []prowconfig.OrgRepo) (*pluginhelp.PluginHelp, error) {
+	pluginHelp := &pluginhelp.PluginHelp{
+		Description: `The publicize plugin is used for merging a pull request into the bonded repository.`,
+	}
+	pluginHelp.AddCommand(pluginhelp.Command{
+		Usage:       "/publicize",
+		Description: "Merge the specific pull request into the bonded repository",
+		WhoCanUse:   "Members of the trusted organization for the repo.",
+		Examples:    []string{"/publicize"},
+	})
+	return pluginHelp, nil
+}
+
+type server struct {
+	webhookTokenGenerator func() []byte
+	githubTokenGenerator  func() []byte
+
+	bonds   map[string]string
+	confirm bool
+
+	gitName  string
+	gitEmail string
+
+	ghc github.Client
+	log *logrus.Entry
+}
+
+func (s *server) ServeHTTP(w http.ResponseWriter, r *http.Request) {
```

Comment:
All of this should be in a library upstream IMO

Reply:
Yea, it seems that this func is duplicated in all the plugins... I will check and open a PR to share it.